### ISSUE TITLE
WIP: How to define the GraphViz record structure?

### ISSUE THREAD

**Issue by clemens-tolboom** [clue/graph-uml#19]
As #9 was just a tiny start let's define how to build this new class.
- [ ] Define a `class GraphVizRecord`
- [ ] Define a subclass for a GraphvizDatabaseRecord structure?
- [ ] Define a subclass for a Class Diagram?
- [ ] Apply the subclass ClassDiagram?

Relates to
- https://github.com/clue/graph/pull/82
- https://github.com/clue/graph/pull/89

**Comment by clemens-tolboom** [comment]
While writing the issue summary I guess starting directly with a `GraphVizRecord` is too quick.

I should reread http://www.graphviz.org/content/node-shapes#record

```
rlabel :== field ( '|' field )*
field :== fieldId or '{' rlabel '}'
fieldId :== [ '<' string '>'] [ string ]
```

Has the usual [escape sequences](http://www.graphviz.org/content/attrs#kescString).
- do we have covered all these in GraphViz?

It also mentions [port position](http://www.graphviz.org/content/attrs#kportPos) to draw an edge to a field within the record.

So maybe we can build the tree using `Graph` with `Layout()` and using subclassing?
- https://github.com/clue/graph/pull/82
- https://github.com/clue/graph/pull/89

**Comment by clue** [comment]
Currently, the `ClassDiagramBuilder` is quite difficult to work with, as it doesn't even remotely resemble a SOLID design :)

I think we should start off by refactoring the code into smaller units:
- Create a new `ClassVertex` that extends the base `Vertex` (depends on clue/graph#82). It should abstract the `Reflection` interface and do little more than that.
- Create a `ClassVertexRecordExporter` (or any better name) that merely operates on a given `ClassVertex` instance in order to produce the required GraphViz dot script.
- Update `ClassDiagramBuilder` to build a `ClassVertex` instead and make sure it's hardly any more than glue code remaining :)
- Consider building a `ClassVertexHtmlExporter` for future use (as per #5)
- Consider how to embed GraphViz port attributes (depends on clue/graph#92)

**Comment by clemens-tolboom** [comment]
Let's close this issue as both graphviz as we think as stated on http://www.graphviz.org/content/node-shapes

> The record-based shape has largely been superseded and greatly generalized by HTML-like labels.

**Comment by clemens-tolboom** [comment]
Subgraphs are requested in  https://github.com/clue/graph/pull/36

**Comment by clue** [comment]
Closed via #23.
